virtcontainers/kata_agent.go

```diff
@@ -1537,7 +1537,11 @@ func (k *kataAgent) handleDeviceBlockVolume(c *Container, device api.Device) (*g
 		vol.Source = blockDrive.DevNo
 	case c.sandbox.config.HypervisorConfig.BlockDeviceDriver == config.VirtioBlock:
 		vol.Driver = kataBlkDevType
-		vol.Source = blockDrive.PCIAddr
+		if blockDrive.PCIAddr == "" {
+			vol.Source = blockDrive.VirtPath
```

Comment:
So looks like we are using device names predicted on the host side instead of PCI addresses. While this may work, @sboeuf I do recall we had seen issues in the past due to this approach with qemu.

Reply:
@amshinde I agree but I think this is good enough for right now. As discussed offline with @likebreath, we can add to Cloud-Hypervisor a simple payload returned from the hotplug, providing the information about PCI b/d/f. Unfortunately it's not there yet, which is why in the meantime, this solution sounds like a good tradeoff.
/cc @sameo